src/utilities/util_dump.c

```diff
@@ -814,6 +847,321 @@ dump_all_records(WT_CURSOR *cursor, bool reverse, bool json)
     return (0);
 }
 
+/*
+ * dump_explore --
+ *     Dump data in an interactive fashion.
+ */
+static int
+dump_explore(WT_CURSOR *cursor, const char *uri, bool reverse, bool pretty, bool hex, bool json)
+{
+#define MAX_LIMIT 256
+    WT_DECL_RET;
+    WT_SESSION *session;
+    const char *infix, *key, *prefix, *suffix, *value;
+    char *args[MAX_LIMIT], *bookmarks[MAX_LIMIT];
+    char *first_arg, user_input[MAX_LIMIT], *current_arg;
+    bool once, search_near;
+    int i, exact, num_args;
+    uint64_t bookmark_index;
+
+    session = cursor->session;
+    once = search_near = false;
+    i = exact = num_args = 0;
+    bookmark_index = 0;
+    for (i = 0; i < MAX_LIMIT; ++i)
+        bookmarks[i] = NULL;
+
+    printf("**************************\n");
+    printf("Explore mode for %s.\n", uri);
+    printf("**************************\n");
+    printf("Enter 'h' for help, 'q' to exit.\n\n");
+
+    if (json) {
+        prefix = "\n{\n";
+        infix = ",\n";
+        suffix = "\n}";
+    } else {
+        prefix = "";
+        infix = "\n";
+        suffix = "\n";
+    }
+
+    while (ret == 0) {
+        i = num_args = 0;
+        fgets(user_input, MAX_LIMIT, stdin);
+
+        /* Remove new line character. */
+        user_input[strlen(user_input) - 1] = '\0';
+        if (strlen(user_input) == 0 && !once)
+            continue;
+        once = true;
+
+        /* Time to exit. */
+        if (strlen(user_input) == 1 && user_input[0] == 'q')
+            break;
+
+        /* Parse the input. */
+        current_arg = strtok(user_input, " ");
+        while (current_arg != NULL) {
+            args[i] = malloc(MAX_LIMIT);
+            strcpy(args[i++], current_arg);
+            ++num_args;
+            current_arg = strtok(NULL, " ");
+        }
+
+        first_arg = args[0];
+        switch (first_arg[0]) {
+        /* Cursor info. */
+        case 'a':
+            ret = cursor->get_key(cursor, &key);
+            if (ret != 0 && ret != EINVAL)
+                return (util_cerr(cursor, "get_key", ret));
+            else if (ret == EINVAL) {
+                printf("Error: the cursor needs to be positioned.\n");
```

Comment:
I'm curious in which case(s) one could run into an unpositioned cursor. Will running cursor->next to the end be one of the cases? Should we add the suggestion of running cursor reset "-c" in the error message?

Reply:
This case happens when the first command is `a` or whenever the cursor is reset and the next command is `a`.